lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -49,6 +49,12 @@ def as_test_response(obj)
               resp.request = ActionDispatch::Request.new({})
             end
             ::ActionDispatch::TestResponse.from_response(obj)
+          elsif obj.respond_to?(:status) && obj.respond_to?(:headers) && obj.respond_to?(:body)
+            # Hack to support `Rack::MockResponse` without having to load
+            # Rack or catch `NameError`s for the undefined constants
+            resp = ::ActionDispatch::Response.new(obj.status, obj.headers, obj.body)
+            resp.request = ::ActionDispatch::Request.new({})
```

Comment:
What was your intent with this? Its discarded immediately and isn't necessary to pass the spec you've written.

Reply:
You're right.

I initially wrote it a lot like the above condition to support Capybara Session. It looked like this:
```ruby
obj = ActionDispatch::Response.new.tap do |resp|
  resp.status = obj.status
  resp.headers.clear
  resp.headers.merge!(obj.headers)
  resp.body = obj.body
  resp.request = ActionDispatch::Request.new({})
end
::ActionDispatch::TestResponse.from_response(obj)
```

Then I refactored to use the initializer:

```ruby
obj = ActionDispatch::Response.new(obj.status, obj.headers, obj.body).tap do |resp|
  resp.request = ActionDispatch::Request.new({})
end
::ActionDispatch::TestResponse.from_response(obj)
```

Then I got rid of the `tap` which does not add much and renamed the variable:
```ruby
resp = ActionDispatch::Response.new(obj.status, obj.headers, obj.body)
resp.request = ActionDispatch::Request.new({})
::ActionDispatch::TestResponse.from_response(obj)
```

And I forgot to rename `obj` to `resp` in the last line.

As it works that way, I'll simplify the code.